engine/classes/Elgg/Application.php

```diff
@@ -333,6 +333,15 @@ public function __get($name) {
 	 * @return self
 	 */
 	private static function create() {
+		// we need to register for shutdown before Symfony registers the
+		// session_write_close() function. https://github.com/Elgg/Elgg/issues/9243
+		register_shutdown_function(function () {
+			// There are cases where we may exit before this function is defined.
+			if (function_exists('_elgg_shutdown_hook')) {
+				_elgg_shutdown_hook();
+			}
+		});
+
```

Comment:
@mrclay Is this how you intended 2.0 to handle https://github.com/Elgg/Elgg/blob/1.12/engine/start.php#L31